### DISCUSSION TITLE
联合查询是否可新增类型转换

### DISCUSSION THREAD

**Discussion by exy3351** [dotnetcore/FreeSql#1792] [Ideas]
需求： 综合分页查询多个表(不是联表查询或子查询)，表的字段不完全一致，需要做类型映射

实体 RowItem 最后查询后返回的对象
public class RowItem{
        public string Zl {set;set}
}

实体 ZRZ 要查询的表1 ，其中字段 Lzzl 需要映射到 RowItem的Zl上
public class ZRZ{
      public string Lzzl{set;set}
}

实体 H 要查询的表2
public class H{
      public string zl {set;set}
}

对应的sql语句

select Zl  from(
     select Lzzl zl from ZRZ where Lzzl  like '%1%'
     union all
     select zl from Hwhere zl  like '%1%'
) b

期望C# 代码可以实现类似的类型转换

fsql.Select<RowItem>()
    .UnionAll(
        fsql.Select<ZRZ>().Where(a => a.Lzzl.Contains(condition.Zl)).AsType<RowItem>(a =>
        {
            Zl = a.Lzzl
        }),
        fsql.Select<H>().Where(a => a.Zl.Contains(condition.Zl)).AsType<RowItem>()
    )
    .Page(1,20);

**Comment by 2881099** [comment]
看下嵌套查询文档

**Reply by exy3351** [reply]
这个文档看过了的，但好像是不满足，多表查询，只有union all ，没有join，文档中好像都与join有关。
大概的场景是 分页查询 A、B、C、D... 多个没有直接关系的表，将查询结果 拼成一个公共的对象，如只有 名称、地址等，但是每个表的字段名可能不一样，需要转成公共对象中的名称、地址

**Reply by 2881099** [reply]
看一下联合查询文档，配合嵌套查询
